### PR TITLE
[RSN-31] - Create validators according to constraints

### PR DESCRIPTION
**Pull request by mkoper02**
## Description
Create validators which will validate entities on backend side before putting them into database.

## Related issue and/or ticket
[RSN-31](https://toopricey.youtrack.cloud/issue/RSN-31/BackEnd-Create-validators-according-to-constraints)

### CODE REVIEW

**Review by wzarek — CHANGES_REQUESTED** [review]
Regexe strings and numbers can be put as a seperate const value in class itself (will be cleaner that way and easier to do changes).

**Comment by wzarek** [comment]
skill issues, assigning @raczu

File: Server/ReasnAPI/ReasnAPI/Controllers/AuthController.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
+using FluentValidation;
+using Microsoft.AspNetCore.Mvc;
+using ReasnAPI.Mappers;
+using ReasnAPI.Models.Authentication;
+using ReasnAPI.Services.Authentication;
+
+namespace ReasnAPI.Controllers;
+
+[ApiController]
+[Route("api/[controller]")]
```
Comment (wzarek) [inline Server/ReasnAPI/ReasnAPI/Controllers/AuthController.cs:10]: Maybe do some global route prefix in Program.cs to not write `api` all the time, wdyt?

**Review by wzarek — APPROVED** [review]
@raczu - please fix conflicts / rebase
Also check comment, other than that lgtm (decide if worth doing, if so - commit and merge, if not - just merge)

File: Server/ReasnAPI/ReasnAPI/Controllers/AuthController.cs (hunk 1 above)
Comment (raczu) [reply Server/ReasnAPI/ReasnAPI/Controllers/AuthController.cs:10]: This will be applied in #43.

**Pull request merged**